src/mlpack/methods/ne/neuron_gene.hpp

```diff
@@ -0,0 +1,246 @@
+/**
+ * @file neuron_gene.hpp
+ * @author Bang Liu
+ *
+ * Definition of the NeuronGene class.
+ */
+#ifndef MLPACK_METHODS_NE_NEURON_GENE_HPP
+#define MLPACK_METHODS_NE_NEURON_GENE_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/activation_functions/logistic_function.hpp>
+#include <mlpack/methods/ann/activation_functions/rectifier_function.hpp>
+#include <mlpack/methods/ann/activation_functions/tanh_function.hpp>
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This is enumeration of neuron types.
+ */
+enum NeuronType
+{
+  NONE = 0,
+  INPUT,
+  BIAS,
+  HIDDEN,
+  OUTPUT
+};
+
+/**
+ * This is enumeration of activation function types.
+ */
+enum ActivationFuncType
+{
+    SIGMOID = 0,
+    TANH,
+    LINEAR,
+    RELU,
+
+    COUNT  //  Used to count how many activation types.
+};
+
+/**
+ * This class defines a neuron gene.
+ */
+class NeuronGene
+{
+ public:
+  /**
+   * Default constructor.
+   */
+  NeuronGene() {}
+  
+  /**
+   * Parametric constructor.
+   *
+   * @param id Neuron's ID.
+   * @param type Neuron's type: INPUT, BIAS, HIDDEN, OUTPUT.
+   * @param actFuncType The type of activation function.
+   * @param depth The depth of the neuron.
+   * @param coordinate The coordinate of the neuron.
+   * @param input Neuron's input value.
+   * @param activation Neuron's activation value.
+   */
+  NeuronGene(int id,
+  	         NeuronType type,
+  	         ActivationFuncType actFuncType,
+             double depth,
+             std::vector<double> coordinate,
```

Comment:
I think it would be a good idea to provide another constructor, that uses a default value for ``coordinate``. So that I can use:

```
NeuronGene inputGene1(0, INPUT, LINEAR, 0, 0, 0);
```

instead of:

```
NeuronGene inputGene1(0, INPUT, LINEAR, 0, std::vector<double>(), 0, 0);
```